src/Cursor.js

```diff
@@ -69,6 +70,15 @@ var cursorBuildMemoizer = util.memoizeFactory(function (cmp, path, value) {
 var build = cursorBuildMemoizer(function (cmp, path, value) {
   path = path === undefined ? [] : path;
   value = value || util.getRefAtPath(cmp.state, path);
+
+  if (_.isObject(value)) {
+    Object.observe(value, function (changes) {
```

Comment:
Object.observe is cool, but there is not much browsers supporting it. We could better to leverage [Object.freeze](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze), which is widely used in React and more browsers supporting it.